core/services/fluxmonitorv2/orm_test.go

```diff
@@ -0,0 +1,251 @@
+
+	orm := fluxmonitorv2.NewORM(corestore.DB)
+
+	address := cltest.NewAddress()
+
+	for round := uint32(0); round < 10; round++ {
+		_, err := orm.FindOrCreateFluxMonitorRoundStats(address, round)
+		require.NoError(t, err)
+	}
+
+	count, err := orm.CountFluxMonitorRoundStats()
+	require.NoError(t, err)
+	require.Equal(t, 10, count)
+
+	err = orm.DeleteFluxMonitorRoundsBackThrough(cltest.NewAddress(), 5)
+	require.NoError(t, err)
+
+	count, err = orm.CountFluxMonitorRoundStats()
+	require.NoError(t, err)
+	require.Equal(t, 10, count)
+
+	err = orm.DeleteFluxMonitorRoundsBackThrough(address, 5)
+	require.NoError(t, err)
+
+	count, err = orm.CountFluxMonitorRoundStats()
+	require.NoError(t, err)
+	require.Equal(t, 5, count)
+}
+
+func TestORM_UpdateFluxMonitorRoundStats(t *testing.T) {
+	t.Parallel()
+
+	corestore, cleanup := cltest.NewStore(t)
+	t.Cleanup(cleanup)
+
+	// Instantiate a real pipeline ORM because we need to create a pipeline run
+	// for the foreign key constraint of the stats record
+	eventBroadcaster := postgres.NewEventBroadcaster(
+		corestore.Config.DatabaseURL(),
+		corestore.Config.DatabaseListenerMinReconnectInterval(),
+		corestore.Config.DatabaseListenerMaxReconnectDuration(),
+	)
+	pipelineORM := pipeline.NewORM(corestore.ORM.DB, corestore.Config, eventBroadcaster)
+	// Instantiate a real job ORM because we need to create a job to satisfy
+	// a check in pipeline.CreateRun
+	jobORM := job.NewORM(corestore.ORM.DB, corestore.Config, pipelineORM, eventBroadcaster, &postgres.NullAdvisoryLocker{})
+	orm := fluxmonitorv2.NewORM(corestore.DB)
+
+	address := cltest.NewAddress()
+	var roundID uint32 = 1
+
+	j := makeJob(t)
+	err := jobORM.CreateJob(context.Background(), j, *pipeline.NewTaskDAG())
+	require.NoError(t, err)
+
+	for expectedCount := uint64(1); expectedCount < 4; expectedCount++ {
+		runID, err := pipelineORM.CreateRun(context.Background(), j.ID, map[string]interface{}{})
+		require.NoError(t, err)
+
+		err = orm.UpdateFluxMonitorRoundStats(address, roundID, runID)
+		require.NoError(t, err)
+
+		stats, err := orm.FindOrCreateFluxMonitorRoundStats(address, roundID)
+		require.NoError(t, err)
+		require.Equal(t, expectedCount, stats.NumSubmissions)
+		require.True(t, stats.PipelineRunID.Valid)
+		require.Equal(t, runID, stats.PipelineRunID.Int64)
+	}
+}
+
+func makeJob(t *testing.T) *job.SpecDB {
+	t.Helper()
+
+	return &job.SpecDB{
+		IDEmbed:       job.IDEmbed{ID: 1},
+		Type:          "fluxmonitor",
+		SchemaVersion: 1,
+		Pipeline:      *pipeline.NewTaskDAG(),
+		FluxMonitorSpec: &job.FluxMonitorSpec{
+			IDEmbed:           job.IDEmbed{ID: 2},
+			ContractAddress:   cltest.NewEIP55Address(),
+			Precision:         2,
+			Threshold:         0.5,
+			PollTimerPeriod:   1 * time.Second,
+			PollTimerDisabled: false,
+			IdleTimerPeriod:   1 * time.Minute,
+			IdleTimerDisabled: false,
+			CreatedAt:         time.Now(),
+			UpdatedAt:         time.Now(),
+		},
+	}
+}
+
+func TestORM_CreateEthTransaction(t *testing.T) {
```

Comment:
hmm this is actually identical to the OCR tests + transmitter right? Could we just pull out a ethtransmitter package and use it in both places?

Reply:
cc @samsondav what do you think about that? Feels strange to duplicate this

Reply:
There is a slight different where this implementation is using gorm, while OCR uses sql.DB. We could look at extracting the code out though